languages/csharp/exercises/concept/parameters/.meta/Example.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+
+namespace example
+{
+    public class RemoteControlCar
+    {
+        private int batteryPercentage = 100;
+        private int distanceDrivenInMeters = 0;
+        private string[] sponsors = new string[0];
+        private long latestTimestamp = 0L;
+
+        public void Drive()
+        {
+            if (batteryPercentage > 0)
+            {
+                batteryPercentage -= 10;
+                distanceDrivenInMeters += 2;
+            }
+        }
+
+        public void AddSponsors(params string[] sponsors)
+        {
+            this.sponsors = sponsors;
+        }
+
+        public string DisplaySponsor(int sponsorNum)
+        {
+            return sponsors[sponsorNum];
+        }
+
+        public bool GetTelemetryData(ref long timestamp,
+            out int batteryPercentage, out int distanceDrivenInMeters)
+        {
+            if (timestamp > latestTimestamp)
+            {
+                latestTimestamp = timestamp;
+                batteryPercentage = this.batteryPercentage;
+                distanceDrivenInMeters = this.distanceDrivenInMeters;
+                return true;
+            }
+            else
+            {
+                timestamp = latestTimestamp;
+                batteryPercentage = -1;
+                distanceDrivenInMeters = -1;
+                return false;
+            }
+        }
+
+        public static RemoteControlCar Buy()
+        {
+            return new RemoteControlCar();
+        }
+    }
+
+    public class TelemetryClient
+    {
+        private RemoteControlCar car;
+
+        public TelemetryClient(RemoteControlCar car)
+        {
+            this.car = car;
+        }
+
+        public string IsCarOk(long timestamp)
```

Comment:
Is there anything this method provides that the `GetBatteryUsagePerMeter` doesn't? Except for discarding the value. If not, maybe we could remove this method and just have the other method to reduce the number of representations.

Reply:
Ok let's discard discard.

It's no great loss and covered in _after.md_